src/ffi/lib_ffi.cr

```diff
@@ -0,0 +1,84 @@
+require "./enums"
+
+@[Link("ffi")]
+lib LibFFI
```

Comment:
Similar to readline, we could hide it in Crystal::. But maybe there are some benefits on leaving this in the std-lib.

But most important this is adding a new dependency I'm not sure is always available.
Should we have an opt-out compiler flag that will disable the repl and remove the need of ffi and readline?

Reply:
Let's hide it in the Crystal namespace for now.

Should there be a `-D` flag names `--with-interpreter` or `--without-interpreter`? I'm not sure whether the interpreter should be included by default or not. libffi and readline are pretty ubiquitous, so it would be easy to include them in every Crystal distribution.

Reply:
I think it's better an opt-out `--without-interpreter`. If something breaks package maintainers can remove the feature and are aware of that.

If we do opt-in then users will need to report maintainers about the missing feature in order to use it.

Reply:
For a start the CI fails because `readline` is not present 🙈 

>    /usr/bin/ld: cannot find -lreadline (this usually means you need to install the development package for libreadline)

Reply:
Compiler dependencies don't need to be included in Crystal distributions. We don't ship LLVM in our distribution packages.

The compiler binary should obviously have these libraries linked statically. But we shouldn't ship the libraries separately in the default distribution packages.

They can obviously be included in the build packages. Otherwise, if you want to build the compiler (with these features), you need to install the dependencies. Most Crystal users don't want to build the compiler.

So, there should be flags similar to `without_zlib` and `without_openssl` to disable the features that would link against these libraries (i.e. `crystal i`).

Reply:
How can I install readline on CI? Also feel free to push commits in the branch

Maybe I can move the branch to this repo so it's easier to contribute to?

Reply:
After 1.1 release we can rebuild those images to include manually readline on those and be sure next images will install it.